File: edumanage/springboot/src/main/java/com/keshe/edumanage/entity/exam/ExamMonitor.java
package com.keshe.edumanage.entity.exam;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.keshe.edumanage.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;
import java.time.LocalTime;

@Data
@TableName("exam_monitor")
@EqualsAndHashCode(callSuper = true)
public class ExamMonitor extends BaseEntity {
    private Long examId;
    private Long teacherId;

    /**
     * 主监考/副监考
     */
    private String monitorRole;

    /**
     * 以下为展示用字段（不属于 exam_monitor 表），
     * 由 ExamMonitorService 关联 exam_info / exam_room / base_classroom / base_teacher 填充。
     */
    @TableField(exist = false)
    private String examName;

    @TableField(exist = false)
    private String courseName;

    @TableField(exist = false)
    private LocalDate examDate;

    @TableField(exist = false)
    private LocalTime startTime;

    @TableField(exist = false)
    private LocalTime endTime;

    /**
     * 考场名称（同一场考试有多个考场时以顿号连接）
     */
    @TableField(exist = false)
    private String roomName;

    @TableField(exist = false)
    private String teacherName;
}

File: student-teacher/src/api/examMonitor.ts
import { http, normalizeList } from '@/utils/request'

/** 监考安排（exam_monitor.monitor_role：MAIN 主监考 / SUB 副监考） */
export interface ExamMonitor {
  id?: number
  examId: number
  teacherId: number
  monitorRole?: string
  /** 展示用扩展字段（后端实体没有，Mock 会带） */
  examName?: string
  courseName?: string
  examDate?: string
  startTime?: string
  endTime?: string
  roomName?: string
  className?: string
}

/**
 * 教师：我的监考任务
 * 后端 GET /api/exam-monitor/list-by-teacher/{teacherId}
 */
export async function listMyInvigilations(teacherId: number): Promise<ExamMonitor[]> {
  const data = await http.get<unknown>(`/exam-monitor/list-by-teacher/${teacherId}`)
  return normalizeList<ExamMonitor>(data)
}

/**
 * 查询某场考试的监考安排
 * 后端 GET /api/exam-monitor/list-by-exam/{examId}
 */
export async function listInvigilationsByExam(examId: number): Promise<ExamMonitor[]> {
  const data = await http.get<unknown>(`/exam-monitor/list-by-exam/${examId}`)
  return normalizeList<ExamMonitor>(data)
}
